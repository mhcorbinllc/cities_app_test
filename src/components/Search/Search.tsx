import React, { useState } from 'react'
import { Form, FacnyInput, ResultsTable, Spinner } from '../Common/Index'
import useCityDataAndFilter from '../../hooks/useCityDataAndFilter'
import { faSearch } from '@fortawesome/free-solid-svg-icons'

import './Search.scss'

const Search = () => {
    const [searchString, setSearchString] = useState("")
    const { isLoading, data } = useCityDataAndFilter(searchString)

    return (
        <div className="search-container">
            { !isLoading ? <div>
                <Form aria-label="search-form" sticky>
                    <FacnyInput 
                        value={searchString} 
                        onChange={(e) => setSearchString(e.target.value)}
                        icon={faSearch} 
                        iconColor="#fff"
                        type="text" 
                        placeholder="Search by city name..." 
                        aria-label="search-field"/>
                </Form>
                { data.length > 0 && <ResultsTable headers={["State", "City"]} data={data} /> }
            </div> : <Spinner color="blue" size="md" />}
        </div>
    )
}

export default Search