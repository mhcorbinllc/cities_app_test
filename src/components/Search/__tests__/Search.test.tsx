import "jest";

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })

import React from 'react'
import Search from '../Search';

describe('Search', () => {
    let wrapper

    test('renders without Error and with no search results visible', () => {
        wrapper = Enzyme.mount(<Search />)

        expect(wrapper).toMatchSnapshot()

        expect(wrapper.find('input').length).toEqual(1)
        //no table because no search results
        expect(wrapper.find('table').length).toEqual(0)
    })

    test ('renders search results when search parm is provided', () => {
        wrapper = Enzyme.mount(<Search />)

        wrapper.find('input').simulate('change', { target: { value: "New York" } })

        //table exists because search results returned
        expect(wrapper.find('table').length).toEqual(1)

        //two rows container New York
        expect(wrapper.find('tbody tr').length).toEqual(2)
    }) 
})