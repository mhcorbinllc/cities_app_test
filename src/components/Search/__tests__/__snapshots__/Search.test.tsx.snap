// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders without Error 1`] = `
<Search>
  <div
    className="search-container"
  >
    <div>
      <Form
        aria-label="search-form"
        sticky={true}
      >
        <div
          className="form-container sticky"
        >
          <form
            className="form-primary"
          >
            <FancyInput
              aria-label="search-field"
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
              iconColor="#fff"
              onChange={[Function]}
              placeholder="Search by city name..."
              type="text"
              value=""
            >
              <div
                className="fancy-input-container"
              >
                <input
                  aria-label="search-field"
                  onChange={[Function]}
                  placeholder="Search by city name..."
                  type="text"
                  value=""
                />
                <FontAwesomeIcon
                  border={false}
                  className=""
                  color="#fff"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-search fa-w-16 "
                    color="#fff"
                    data-icon="search"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </div>
            </FancyInput>
          </form>
        </div>
      </Form>
    </div>
  </div>
</Search>
`;

exports[`Search renders without Error and with no search results visible 1`] = `
<Search>
  <div
    className="search-container"
  >
    <div>
      <Form
        aria-label="search-form"
        sticky={true}
      >
        <div
          className="form-container sticky"
        >
          <form
            className="form-primary"
          >
            <FancyInput
              aria-label="search-field"
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
              iconColor="#fff"
              onChange={[Function]}
              placeholder="Search by city name..."
              type="text"
              value=""
            >
              <div
                className="fancy-input-container"
              >
                <input
                  aria-label="search-field"
                  onChange={[Function]}
                  placeholder="Search by city name..."
                  type="text"
                  value=""
                />
                <FontAwesomeIcon
                  border={false}
                  className=""
                  color="#fff"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-search fa-w-16 "
                    color="#fff"
                    data-icon="search"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </div>
            </FancyInput>
          </form>
        </div>
      </Form>
    </div>
  </div>
</Search>
`;

exports[`Search renders without Error and without any search results visible 1`] = `
<Search>
  <div
    className="search-container"
  >
    <div>
      <Form
        aria-label="search-form"
        sticky={true}
      >
        <div
          className="form-container sticky"
        >
          <form
            className="form-primary"
          >
            <FancyInput
              aria-label="search-field"
              icon={
                Object {
                  "icon": Array [
                    512,
                    512,
                    Array [],
                    "f002",
                    "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                  ],
                  "iconName": "search",
                  "prefix": "fas",
                }
              }
              iconColor="#fff"
              onChange={[Function]}
              placeholder="Search by city name..."
              type="text"
              value=""
            >
              <div
                className="fancy-input-container"
              >
                <input
                  aria-label="search-field"
                  onChange={[Function]}
                  placeholder="Search by city name..."
                  type="text"
                  value=""
                />
                <FontAwesomeIcon
                  border={false}
                  className=""
                  color="#fff"
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        512,
                        512,
                        Array [],
                        "f002",
                        "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                      ],
                      "iconName": "search",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-search fa-w-16 "
                    color="#fff"
                    data-icon="search"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </div>
            </FancyInput>
          </form>
        </div>
      </Form>
    </div>
  </div>
</Search>
`;
